netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -195,8 +193,13 @@ private void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
           checkNotNull(transportListener.methodDetermined(method, metadata), "statsTraceCtx");
       NettyServerStream.TransportState state = new NettyServerStream.TransportState(
           this, http2Stream, maxMessageSize, statsTraceCtx);
-      NettyServerStream stream = new NettyServerStream(ctx.channel(), state, attributes,
+      Attributes attributesWithAuthority = Attributes
+              .newBuilder()
+              .setAll(attributes).set(ServerImpl.ATTR_AUTHORITY, headers.authority().toString())
```

Comment:
This is going to add extra allocation and AsciiString conversion in the serving path, where there was none before.  Almost no users need this data.  This would slow down all servers for the sake of proxies.